sky/clouds/service_catalog/aws_catalog.py

```diff
@@ -148,14 +149,19 @@ def _fetch_and_apply_az_mapping(df: pd.DataFrame) -> pd.DataFrame:
 
 
 def _get_df() -> pd.DataFrame:
-    if config.get_use_default_catalog():
-        return _default_df
-    else:
-        global _user_df
-        with _apply_az_mapping_lock:
-            if _user_df is None:
+    global _user_df
+    with _apply_az_mapping_lock:
+        if _user_df is None:
```

Comment:
In the failure case, L157+, does that mean the next call trigger the fetch and exception again? Shall we prevent that? I guess it depends on how many times _get_df() is called per run.

Reply:
That is a good point! The failure should be a edge case, where a normal user may not encounter. I checked the `sky status` and `sky check` for an abnormal account, and seems the latency is not noticable.

Also, It may not be same to save the `_default_df` to `_user_df`, as we don't want to acidentally have some write operators to use the `_default_df`.

I guess it should be fine to query every time?